src/test/integration/IntegrationBase.t.sol

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "forge-std/Test.sol";
+
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import "src/test/integration/IntegrationDeployer.t.sol";
+import "src/test/integration/Global.t.sol";
+import "src/test/integration/users/User.sol";
+
+abstract contract IntegrationBase is IntegrationDeployer {
+
+    /**
+     * Gen/Init methods:
+     */
+
+    /**
+     * @dev Create a new user according to configured random variants.
+     * This user is ready to deposit into some strategies and has some underlying token balances
+     */
+    function _newStaker() internal returns (User, IStrategy[] memory, uint[] memory) {
+        (User staker, IStrategy[] memory strategies, uint[] memory tokenBalances) = _randUser();
+        
+        assert_HasUnderlyingTokenBalances(staker, strategies, tokenBalances, "_newStaker: failed to award token balances");
+
+        return (staker, strategies, tokenBalances);
+    }
+
+    function _newOperator() internal returns (User, IStrategy[] memory, uint[] memory) {
+        (User operator, IStrategy[] memory strategies, uint[] memory tokenBalances) = _randUser();
+        
+        operator.registerAsOperator();
+        // TODO - deposit into strats
+
+        assert_HasUnderlyingTokenBalances(operator, strategies, tokenBalances, "_newOperator: failed to award token balances");
+        assertTrue(delegationManager.isOperator(address(operator)), "_newOperator: operator should be registered");
+
+        return (operator, strategies, tokenBalances);
+    }
+
+    /** 
+     * Common assertions:
+     */
+
+    function assert_HasNoDelegatableShares(User user, string memory err) internal {
+        (IStrategy[] memory strategies, uint[] memory shares) = 
+            delegationManager.getDelegatableShares(address(user));
+        
+        assertEq(strategies.length, 0, err);
+        assertEq(strategies.length, shares.length, "assert_HasNoDelegatableShares: return length mismatch");
+    }
+
+    function assert_HasUnderlyingTokenBalances(
+        User user, 
+        IStrategy[] memory strategies, 
+        uint[] memory expectedBalances, 
+        string memory err
+    ) internal {
+        for (uint i = 0; i < strategies.length; i++) {
+            IStrategy strat = strategies[i];
+            
+            uint expectedBalance = expectedBalances[i];
+            uint tokenBalance;
+            if (strat == BEACONCHAIN_ETH_STRAT) {
+                tokenBalance = address(user).balance;
+            } else {
+                tokenBalance = strat.underlyingToken().balanceOf(address(user));
+            }
+
+            assertEq(expectedBalance, tokenBalance, err);
+        }
+    }
+
+    function assert_HasNoUnderlyingTokenBalance(User user, IStrategy[] memory strategies, string memory err) internal {
```

Comment:
should this just call `assert_HasUnderlyingTokenBalances` with an array of zeroes? little opportunity for de-duping

Reply:
on it.

Reply:
[6ef08a6](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/6ef08a676fe70389631c4627a9aa2d825a660884)